contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -443,3 +504,133 @@ contract SetConfig is SetUp {
     return bytes32((prefix & prefixMask) | (h & ~prefixMask));
   }
 }
+
+contract NOPsSettlement is SetUp {
+  event NOPsSettledOffchain(address[] payees, uint256[] balances);
+
+  function testSettleNOPsOffchainRevertDueToUnauthorizedCaller() public {
+    (IAutomationRegistryMaster2_3 registry, ) = deployAndConfigureAll(AutoBase.PayoutMode.ON_CHAIN);
+
+    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.OnlyFinanceAdmin.selector));
+    registry.settleNOPsOffchain();
+  }
+
+  function testSettleNOPsOffchainRevertDueToOffchainSettlementDisabled() public {
+    (IAutomationRegistryMaster2_3 registry, ) = deployAndConfigureAll(AutoBase.PayoutMode.OFF_CHAIN);
+
+    vm.prank(registry.owner());
+    registry.disableOffchainPayments();
+
+    vm.prank(FINANCE_ADMIN);
+    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.MustSettleOnchain.selector));
+    registry.settleNOPsOffchain();
+  }
+
+  function testSettleNOPsOffchainSuccess() public {
+    // deploy and configure a registry with OFF_CHAIN payout
+    (IAutomationRegistryMaster2_3 registry, ) = deployAndConfigureAll(AutoBase.PayoutMode.OFF_CHAIN);
+    registry.setPayees(PAYEES);
+
+    uint256[] memory balances = new uint256[](TRANSMITTERS.length);
+    for (uint256 i = 0; i < TRANSMITTERS.length; i++) {
+      balances[i] = 0;
+    }
+
+    vm.startPrank(FINANCE_ADMIN);
+    vm.expectEmit();
+    emit NOPsSettledOffchain(PAYEES, balances);
+    registry.settleNOPsOffchain();
+  }
+
+  function testSettleNOPsOffchainSuccessTransmitterBalanceZeroed() public {
+    // deploy and configure a registry with OFF_CHAIN payout
+    (IAutomationRegistryMaster2_3 registry, ) = deployAndConfigureAll(AutoBase.PayoutMode.OFF_CHAIN);
+    registry.setPayees(PAYEES);
```

Comment:
we could move this to `deployAndConfigureAll` so that we don't have to do this in each test that does transmit()

Reply:
done